### PR TITLE
[Building] allow to build native sql from OAP parent folder

### PR DESCRIPTION
**Pull request by zhouyuan**
Signed-off-by: Yuan Zhou <yuan.zhou@intel.com>

## What changes were proposed in this pull request?

refine  pom to allow building from oap-parent folder

### CODE REVIEW

File: oap-native-sql/core/pom.xml [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
+    <scala.binary.version>2.12</scala.binary.version>
+    <scala.version>2.12.8</scala.version>
+    <build.testJarPhase>none</build.testJarPhase>
+    <build.copyDependenciesPhase>package</build.copyDependenciesPhase>
+    <hadoop.deps.scope>provided</hadoop.deps.scope>
+    <hive.deps.scope>provided</hive.deps.scope>
+    <hive.parquet.group>com.twitter</hive.parquet.group>
+    <parquet.deps.scope>provided</parquet.deps.scope>
+    <jars.target.dir>${project.build.directory}/scala-${scala.binary.version}/jars</jars.target.dir>
+
```
Comment (zhouyuan) [inline oap-native-sql/core/pom.xml:43]: remove blank line

**Comment by zhouyuan** [comment]
@yao531441 can you please help to take a look on this?

File: pom.xml [hunk 2]
```diff
@@ -40,6 +40,8 @@
     <modules>
         <module>oap-common</module>
         <module>oap-cache/oap</module>
+        <module>oap-native-sql/core</module>
+        <module>oap-data-source/arrow</module>
```
Comment (zhouyuan) [inline pom.xml:44]: @zhztheplayer  does this looks good to you? I'm trying to make arrow data source to be compiled under OAP folder

Comment (zhztheplayer) [reply pom.xml:44]: Yes looks good to me.

Comment (zhouyuan) [reply pom.xml:44]: @zhztheplayer  thanks!

File: oap-native-sql/core/pom.xml [hunk 3]
```diff
@@ -1,24 +1,45 @@
...
   <modelVersion>4.0.0</modelVersion>
+
   <parent>
-    <groupId>com.intel.sparkColumnarPlugin</groupId>
-    <artifactId>sparkColumnarPlugin</artifactId>
-    <version>1.0</version>
-    <relativePath>../pom.xml</relativePath>
+    <groupId>com.intel.oap</groupId>
+    <artifactId>oap-parent</artifactId>
+    <version>0.9.0</version>
```
Comment (zhouyuan) [inline oap-native-sql/core/pom.xml:21]: @zhztheplayer I'm following OAP release version here, should we also bump the version in oap-data-source?

File: oap-data-source/arrow/pom.xml [hunk 4]
```diff
@@ -19,6 +19,10 @@
       <name>Scala-Tools Maven2 Repository</name>
       <url>http://scala-tools.org/repo-releases</url>
     </repository>
+    <repository>
+      <id>ossrh</id>
+      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+    </repository>
```
Comment (zhouyuan) [inline oap-data-source/arrow/pom.xml:25]: remove this repo and use spark 3.0.0 jar files from maven central

File: oap-data-source/arrow/pom.xml [hunk 5]
```diff
@@ -61,7 +70,7 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.spark</groupId>
+      <groupId>com.intel.spark</groupId>
```
Comment (zhouyuan) [inline oap-data-source/arrow/pom.xml:73]: switch to spark 3.0.0

File: oap-native-sql/core/src/main/java/com/intel/oap/datasource/VectorizedParquetArrowReader.java [hunk 6]
```diff
@@ -70,7 +70,7 @@
 
   public VectorizedParquetArrowReader(String path, ZoneId convertTz, boolean useOffHeap,
       int capacity, StructType sourceSchema, StructType readDataSchema, String tmp_dir) {
-    super(convertTz, useOffHeap, capacity);
+    super(convertTz, "", useOffHeap, capacity);
```
Comment (zhouyuan) [inline oap-native-sql/core/src/main/java/com/intel/oap/datasource/VectorizedParquetArrowReader.java:73]: there's a tiny API change in 3.0.0, seems passing "" can work, the TPCH queries are functional and results are identical. 
 @xuechendi

**Comment by zhouyuan** [comment]
I modified the version to 0.9.0 to match with OAP release @zhztheplayer @xuechendi

File: oap-data-source/arrow/pom.xml [hunk 7]
```diff
@@ -2,12 +2,12 @@
...
   <groupId>com.intel.oap</groupId>
   <artifactId>spark-arrow-datasource</artifactId>
-  <version>0.1.0-SNAPSHOT</version>
+  <version>0.9.0</version>
   <inceptionYear>2008</inceptionYear>
   <properties>
     <scala.version>2.12.10</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
-    <spark.version>3.1.0-SNAPSHOT</spark.version>
+    <spark.version>3.0.0</spark.version>
```
Comment (zhztheplayer) [inline oap-data-source/arrow/pom.xml:10]: Thanks!

**Pull request merged**